### ISSUE TITLE
False positive on the `Style/HashEachMethods` cop for array destructuring

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12444]
## Actual behavior

Before autocorrection

```ruby
def my_method
  [[1, 2, 3], [4, 5, 6]].each { |a, _| puts(a) }
end
```
output:
```console
2.7.6 :002 > [[1, 2, 3], [4, 5, 6]].each { |a, _| puts(a) }
1
4
 => [[1, 2, 3], [4, 5, 6]] 
```

After autocorrection

```ruby
def my_method
  [[1, 2, 3], [4, 5, 6]].each_key { |a| puts(a) }
end
```

output:
```console
2.7.6 :003 > [[1, 2, 3], [4, 5, 6]].each_key { |a| puts(a) }
Traceback (most recent call last):
        4: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `<main>'
        3: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `load'
        2: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):3
NoMethodError (undefined method `each_key' for [[1, 2, 3], [4, 5, 6]]:Array)
Did you mean?  each_entry
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by postmodern** [comment]
Also came here after getting this blatant false positive:

```
lib/ronin/support/crypto/cert.rb:407:13: C: [Correctable] Style/HashEachMethods: Use each_key instead of each and remove the unused (value,critical) block argument.
10
            extensions.each do |name,(value,critical)| ...
11
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

```ruby
            extensions.each do |name,(value,critical)|
              ext = extension_factory.create_extension(name,value,critical)
              cert.add_extension(ext)
            end
```

**Comment by pbstriker38** [comment]
False positive here too
```ruby
[{ 'name' => 'foo' }, { 'name' => 'bar' }].each do |attributes, name: attributes['name']|
  do_something_with_name(name)
end
```
